### PR TITLE
add(): adding prometheus kafka exporter

### PR DESCRIPTION
**Pull request by gkarthiks**
Signed-off-by: gkarthiks <github.gkarthiks@gmail.com>

#### What this PR does / why we need it:
A prometheus metrics exporter for kafka

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

**Comment by maver1ck** [comment]
Why we're creating new Chart instead of add this as sidecar ?

**Comment by gkarthiks** [comment]
@maver1ck  there's already an incubating chart doing that. This is really for an existing kafka-clusters as container or an existing bare-metal installations.

**Comment by Multiply** [comment]
`annotations` is mentioned in the `README.md` but you're not actually using them. Having the ability to set annotations on both the service and the pod would be appreciated, so we can configure scraping.

**Comment by gkarthiks** [comment]
@Multiply  thanks for pointing that! Added the annotations.

**Comment by gkarthiks** [comment]
@Multiply / @maver1ck  can you please approve this PR?

**Comment by Multiply** [comment]
Looks good to me. Would it make sense to allow both podAnnotations and serviceAnnotations as seen in other charts?

**Comment by gkarthiks** [comment]
@Multiply sorry for the delay. Have added the annotations in the service as well. Could you please review and approve the MR

**Comment by gkarthiks** [comment]
Hi @Multiply @maver1ck @seanknox @scottrigby 
 Can someone approve this PR?

**Review by Multiply — APPROVED** [review]
I'm not sure my approval means much, as I'm not a contributor of any sorts, I was merely interested in the work done here :)

**Comment by gkarthiks** [comment]
@Multiply  gotcha. Thanks though

**Comment by gkarthiks** [comment]
comment for removing stale

File: stable/prometheus-kafka-exporter/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+apiVersion: apps/v1beta2
```
Comment (davidkarlsen) [inline stable/prometheus-kafka-exporter/templates/deployment.yaml]:
```suggestion
apiVersion: apps/v1
```

**Comment by gkarthiks** [comment]
@davidkarlsen  changed as suggested.

**Comment by gkarthiks** [comment]
@davidkarlsen  this test is always failing with the following error message.

```
I0507 02:54:48.497] time="2019-05-07T02:50:35Z" level=error msg="Error Init Kafka Client" source="kafka_exporter.go:184"
I0507 02:54:48.497] panic: kafka: client has run out of available brokers to talk to (Is your cluster reachable?)
```

**Comment by maorfr** [comment]
hey @gkarthiks 

Have you considered hosting this chart and adding it to helm hub?
https://github.com/helm/hub/blob/master/Repositories.md

**Comment by gkarthiks** [comment]
@maorfr I will work on that. Thank you!

**Comment by gkarthiks** [comment]
@maorfr Have created a PR to host in hub https://github.com/helm/hub/pull/102. Please do the needful.

**Comment by gkarthiks** [comment]
closing this PR in favor of helm hub.

**Comment by devsharma1979** [comment]
I dont see any helm chart kafka-exporter chart, do you have any plan to add this to helm hub?

**Comment by gkarthiks** [comment]
@devsharma1979  I don't think the helm-charts repo is upto date. With the introduction of the helm-hub, all the third party charts and recent updates are available there. 

I have the `kafka-exporter` charts under there in my repo. For some reason, just the kafka is not listed in there. Checkout the direct repo link here 
https://github.com/gkarthiks/helm-charts/tree/master/charts/prometheus-kafka-exporter

**Comment by devsharma1979** [comment]
Thanks @gkarthiks. I would use your repo, hope helm-hub would get updated soon.

**Pull request closed without merging**